test/codeu/chat/server/ConversationAccessServerTest.java

```diff
@@ -0,0 +1,465 @@
+ *  corresponding methods as we implement them, so let me know when you push a new commit
+ *  that changes the names of what I have listed here. ~ Sarah Abowitz
+ *
+ *  Replaceable parts:
+ *  - codeu.chat.server.Controller.getPermissionMap(ConversationHeader conversation)
+ *  - codeu.chat.server.Controller.updatePermissionMap(Uuid id, HashMap<Uuid, byte> accessMap)
+ *  - codeu.chat.server.Controller.checkMembership(Uuid id, ConversationHeader conversation)
+ *  - ...server.Controller.demoteOwner(byte access, Uuid target, ConversationHeader conversation)
+ *  - ...server.Controller.addOwner(byte access, Uuid target, ConversationHeader conversation)
+ *  - ...server.Controller.ban(byte access, Uuid target, ConversationHeader conversation)
+ *  - ...server.Controller.addMember(byte access, Uuid target, ConversationHeader conversation)
+ *  - codeu.chat.server.View.getPermissionMapAsString(ConversationHeader conversation)
+ *    
+ */
+
+
+package codeu.chat.server;
+
+import java.util.HashMap;
+
+import static org.junit.Assert.*;
+import org.junit.Test;
+import org.junit.Before;
+
+import codeu.chat.common.BasicController;
+import codeu.chat.common.BasicView;
+import codeu.chat.common.ConversationHeader;
+import codeu.chat.common.Message;
+import codeu.chat.common.User;
+import codeu.chat.util.Uuid;
+
+public final class ConversationAccessServerTest {
+
+  private Model model;
+  private BasicController controller;
+  private BasicView view;
+
+  @Before
+  public void doBefore() {
+    model = new Model();
+    controller = new Controller(Uuid.NULL, model);
+    view = new View(model);
+  }
+
+  @Test
+  public void testAddConversation() {
+
+    final User user = controller.newUser("user");
+
+    assertFalse(
+        "Check that user has a valid reference",
+        user == null);
+
+    final ConversationHeader conversation = controller.newConversation(
+        "conversation",
+        user.id);
+
+    assertFalse(
+        "Check that conversation has a valid reference",
+        conversation == null);
+        
+        
+    assertNotNull(
+         "Check that user has a conversation access map",
+          controller.getPermissionMap(conversation));
+  }
+  
+  @Test
+  public void testGetAndUpdateAccess(){
+    final User creator = controller.newUser("creator");
+    final User user = controller.newUser("user");
+     
+    ConversationHeader conversation = controller.newConversation("chat", creator.id);
+    HashMap<Uuid, Byte> accessMap = controller.getPermissionMap(conversation);
+    
+    byte memberByte = 0b001; // Before testing, this must be the equivalent of 001.
+    
+    accessMap.put(user.id, memberByte);
+    controller.updatePermissionMap(conversation, accessMap);
+    
+    assertFalse(
+        "Check that the allowances can be updated",
+        controller.getPermissionMap(conversation).size()!= 1);
+    
+  }
+  
+  @Test
+  public void testAccessSecurity(){
```

Comment:
This method is very long - about 200 lines. Whenever you find yourself writing a method of this length, consider ways that you might refactor your code to make it smaller so that it's easier to understand. 

When it comes to unit tests specifically, it's considered good practice to write many small, concise test methods, each of which should ideally only test one thing. (To give you an idea, some people feel that one should only have a single assertion per test method.  I'm not that dogmatic about it, but it's a nice goal to aim for when writing tests).